Sources/PointFree/Admin/Index.swift

```diff
@@ -0,0 +1,138 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+private let adminEmails = [
+  "mbw234@gmail.com",
+  "stephen.celis@gmail.com"
+]
+
+func requireAdmin<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data>
+  ) -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data> {
+
+  return { conn in
+    conn
+      |> (adminEmails.contains(get1(conn.data).email.unwrap) ? middleware : redirect(to: .secretHome))
+  }
+}
+
+let adminIndex =
+  requireUser
+    <<< requireAdmin
+    <| writeStatus(.ok)
+    >-> respond(adminIndexView.contramap(lower))
+
+private let adminIndexView = View<(Database.User, Prelude.Unit)> { currentUser, _ in
+  ul([
+    li([
+      a([href(path(to: .admin(.newEpisodeEmail(.show))))], ["Send new episode email"])
+      ])
+    ])
+}
+
+let showNewEpisodeEmailMiddleware =
+  requireUser
+    <<< requireAdmin
+    <| writeStatus(.ok)
+    >-> respond(showNewEpisodeView.contramap(lower))
+
+private let showNewEpisodeView = View<(Database.User, Prelude.Unit)> { currentUser, _ in
+  ul(
+    episodes
+      .sorted(by: ^\.sequence)
+      .map(li <<< newEpisodeEmailRowView.view)
+    )
+}
+
+private let newEpisodeEmailRowView = View<Episode> { ep in
+  p([
+    .text(encode(ep.title)),
+    form([action(path(to: .admin(.newEpisodeEmail(.send(ep.id))))), method(.post)], [
+      input([type(.submit), value("Send email!")])
+      ])
+    ])
+}
+
+let sendNewEpisodeEmailMiddleware: Middleware<StatusLineOpen, ResponseEnded, T2<Episode.Id, Prelude.Unit>, Data> =
+  requireEpisode(notFoundMiddleware: redirect(to: .admin(.newEpisodeEmail(.show))))
+    <<< requireUser
+    <<< requireAdmin
```

Comment:
I think this is a good example of how the tuple type is helping us out. we can stack these transformers on top and come out with a `Tuple3` on the other end (well, kinda. still need to merge my prelude PR to make this a lil better)

Reply:
Cool! I'm still not quite understanding where we use `T4` vs. `Tuple3` and how the combinators should be (and I wonder why PureScript's are the way they are).